src/validator/SocialRecoveryValidator.sol

```diff
@@ -129,7 +131,7 @@ contract SocialRecoveryValidator is IKernelValidator {
                 )
             );
             require(weight > 0, "RecoveryPlugin: weight is zero");
-            isGuardian[msg.sender][guardian]=true;
+            isGuardian[msg.sender][guardian] = true;
```

Comment:
Shouldn’t this be isGuardian[guardian][msg.sender]?
According to 4337 storage access rule, the last key should be the userOp.sender not the first one. It’s quite counter intuitive but that is what we should follow

Reply:
It kinda helped me to add the user defined types for clarity instead of using plain address

In this case, i would define 1) Guardian type 2) Sender type for clarity
https://soliditylang.org/blog/2021/09/27/user-defined-value-types/